src/app/workflow/executions/executions-tab.component.ts

```diff
@@ -169,12 +169,20 @@ export class ExecutionsTabComponent extends EntryTab implements OnChanges {
    */
   private createExecutionsTable(metrics: Metrics | null): ExecutionMetricsTableObject[] {
     let executionsTable: ExecutionMetricsTableObject[] = [];
-    // Only create the table if one of the execution metrics exist
-    if (metrics && (metrics.cpu || metrics.memory || metrics.executionTime || metrics.cost)) {
-      executionsTable.push({ metric: 'CPU', ...metrics?.cpu });
-      executionsTable.push({ metric: 'Memory', ...metrics?.memory });
-      executionsTable.push({ metric: 'Run Time', ...metrics?.executionTime });
-      executionsTable.push({ metric: 'Cost', ...metrics?.cost });
+    // Only add the rows if there are data for that type
+    if (metrics) {
+      if (metrics.cpu) {
+        executionsTable.push({ metric: 'CPU', ...metrics?.cpu });
```

Comment:
The question mark here and on lines 178, 181 and 184 are redundant; you're already verified on line 173 that `metrics` exists.